packages/spor-react/src/button/FloatingActionButton.tsx

```diff
@@ -24,68 +30,77 @@ type FloatingActionButtonProps = BoxProps & {
  *  placement="bottom right"
  * />
  */
-export const FloatingActionButton = ({
-  children,
-  icon,
-  variant,
-  isTextVisible: externalIsTextVisible,
-  placement = "bottom right",
-  ...props
-}: FloatingActionButtonProps) => {
-  const [isTextVisible, setIsTextVisible] = React.useState(
-    externalIsTextVisible !== undefined ? externalIsTextVisible : false
-  );
-  const scrollDirection = useScrollDirection();
-  useEffect(() => {
-    if (externalIsTextVisible !== undefined) {
-      return;
-    }
-    const id = window.setTimeout(
-      () => setIsTextVisible(scrollDirection !== "down"),
-      1000
+export const FloatingActionButton = forwardRef<
+  FloatingActionButtonProps,
+  ComponentWithAs<"a" | "button">
```

Comment:
We might want to add support for the generic case here – like if you want to use an internal link – but it's a bit tricky, so I'll postpone it for later.